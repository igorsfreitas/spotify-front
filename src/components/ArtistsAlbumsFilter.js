import React from 'react'
import { Grid, Paper, TextField } from '@material-ui/core';

class ArtistsAlbumsFilter extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
        <Paper className={this.props.classes.paper}>
          <h3>FILTRO DOS ARTISTAS</h3>
          <Grid container spacing={16}>
            <Grid item lg={3} md={3} sm={6} xs={12}>
              <TextField
                id="artist-open-select"
                select
                label="Artistas"
                className={this.props.classes.textField}
                value={this.props.artist}
                onChange={this.props.handleChange('artist')}
                SelectProps={{
                    MenuProps: {
                      className: this.props.classes.menu,
                    },
                }}
                helperText="Selecione o Artista"
                margin="normal"
                >
                {this.props.renderArtists()}
              </TextField>
            </Grid>
            <Grid item lg={3} md={3} sm={6} xs={12}>
              <TextField
                id="year-open-select"
                select
                disabled={this.props.artistAlbums && this.props.artistAlbums.length<1}
                label="PerÃ­odo"
                className={this.props.classes.textField}
                value={this.props.year}
                onChange={this.props.handleChange('year')}
                SelectProps={{
                    MenuProps: {
                      className: this.props.classes.menu,
                    },
                }}
                helperText="Selecione o Ano"
                margin="normal"
                >
                {this.props.renderYears()}
              </TextField>
            </Grid>
          </Grid>
        </Paper>
    )
  }
}

export default ArtistsAlbumsFilter