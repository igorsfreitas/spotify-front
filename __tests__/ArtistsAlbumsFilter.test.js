import React from 'react'
import { shallow } from 'enzyme'
import ArtistsAlbumsFilter from '../src/components/ArtistsAlbumsFilter'

describe ('Testing Artists Component', ()=>{
    it('should render correctly', ()=>{
        const wrapper = shallow(
            <ArtistsAlbumsFilter 
                classes={{}}
                artist={''}
                handleChange={()=>{}}
                renderArtists={()=>{}}
                artistAlbums={[]}
                year={''}
                renderYears={()=>{}}
            />
        )

        expect(wrapper).toMatchSnapshot()
    })
})