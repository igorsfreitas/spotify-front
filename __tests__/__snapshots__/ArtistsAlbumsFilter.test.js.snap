// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Artists Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistsAlbumsFilter
    artist=""
    artistAlbums={Array []}
    classes={Object {}}
    handleChange={[Function]}
    renderArtists={[Function]}
    renderYears={[Function]}
    year=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h3>
          FILTRO DOS ARTISTAS
        </h3>,
        <WithStyles(Grid)
          container={true}
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={3}
            sm={6}
            xs={12}
          >
            <TextField
              SelectProps={
                Object {
                  "MenuProps": Object {
                    "className": undefined,
                  },
                }
              }
              helperText="Selecione o Artista"
              id="artist-open-select"
              label="Artistas"
              margin="normal"
              required={false}
              select={true}
              value=""
              variant="standard"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={3}
            sm={6}
            xs={12}
          >
            <TextField
              SelectProps={
                Object {
                  "MenuProps": Object {
                    "className": undefined,
                  },
                }
              }
              disabled={true}
              helperText="Selecione o Ano"
              id="year-open-select"
              label="Período"
              margin="normal"
              required={false}
              select={true}
              value=""
              variant="standard"
            />
          </WithStyles(Grid)>
        </WithStyles(Grid)>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FILTRO DOS ARTISTAS",
        },
        "ref": null,
        "rendered": "FILTRO DOS ARTISTAS",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Grid)
              item={true}
              lg={3}
              md={3}
              sm={6}
              xs={12}
            >
              <TextField
                SelectProps={
                  Object {
                    "MenuProps": Object {
                      "className": undefined,
                    },
                  }
                }
                helperText="Selecione o Artista"
                id="artist-open-select"
                label="Artistas"
                margin="normal"
                required={false}
                select={true}
                value=""
                variant="standard"
              />
            </WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
              lg={3}
              md={3}
              sm={6}
              xs={12}
            >
              <TextField
                SelectProps={
                  Object {
                    "MenuProps": Object {
                      "className": undefined,
                    },
                  }
                }
                disabled={true}
                helperText="Selecione o Ano"
                id="year-open-select"
                label="Período"
                margin="normal"
                required={false}
                select={true}
                value=""
                variant="standard"
              />
            </WithStyles(Grid)>,
          ],
          "container": true,
          "spacing": 16,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <TextField
                SelectProps={
                  Object {
                    "MenuProps": Object {
                      "className": undefined,
                    },
                  }
                }
                helperText="Selecione o Artista"
                id="artist-open-select"
                label="Artistas"
                margin="normal"
                required={false}
                select={true}
                value=""
                variant="standard"
              />,
              "item": true,
              "lg": 3,
              "md": 3,
              "sm": 6,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "SelectProps": Object {
                  "MenuProps": Object {
                    "className": undefined,
                  },
                },
                "children": undefined,
                "className": undefined,
                "helperText": "Selecione o Artista",
                "id": "artist-open-select",
                "label": "Artistas",
                "margin": "normal",
                "onChange": undefined,
                "required": false,
                "select": true,
                "value": "",
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <TextField
                SelectProps={
                  Object {
                    "MenuProps": Object {
                      "className": undefined,
                    },
                  }
                }
                disabled={true}
                helperText="Selecione o Ano"
                id="year-open-select"
                label="Período"
                margin="normal"
                required={false}
                select={true}
                value=""
                variant="standard"
              />,
              "item": true,
              "lg": 3,
              "md": 3,
              "sm": 6,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "SelectProps": Object {
                  "MenuProps": Object {
                    "className": undefined,
                  },
                },
                "children": undefined,
                "className": undefined,
                "disabled": true,
                "helperText": "Selecione o Ano",
                "id": "year-open-select",
                "label": "Período",
                "margin": "normal",
                "onChange": undefined,
                "required": false,
                "select": true,
                "value": "",
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h3>
            FILTRO DOS ARTISTAS
          </h3>,
          <WithStyles(Grid)
            container={true}
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
              lg={3}
              md={3}
              sm={6}
              xs={12}
            >
              <TextField
                SelectProps={
                  Object {
                    "MenuProps": Object {
                      "className": undefined,
                    },
                  }
                }
                helperText="Selecione o Artista"
                id="artist-open-select"
                label="Artistas"
                margin="normal"
                required={false}
                select={true}
                value=""
                variant="standard"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              lg={3}
              md={3}
              sm={6}
              xs={12}
            >
              <TextField
                SelectProps={
                  Object {
                    "MenuProps": Object {
                      "className": undefined,
                    },
                  }
                }
                disabled={true}
                helperText="Selecione o Ano"
                id="year-open-select"
                label="Período"
                margin="normal"
                required={false}
                select={true}
                value=""
                variant="standard"
              />
            </WithStyles(Grid)>
          </WithStyles(Grid)>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FILTRO DOS ARTISTAS",
          },
          "ref": null,
          "rendered": "FILTRO DOS ARTISTAS",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Grid)
                item={true}
                lg={3}
                md={3}
                sm={6}
                xs={12}
              >
                <TextField
                  SelectProps={
                    Object {
                      "MenuProps": Object {
                        "className": undefined,
                      },
                    }
                  }
                  helperText="Selecione o Artista"
                  id="artist-open-select"
                  label="Artistas"
                  margin="normal"
                  required={false}
                  select={true}
                  value=""
                  variant="standard"
                />
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
                lg={3}
                md={3}
                sm={6}
                xs={12}
              >
                <TextField
                  SelectProps={
                    Object {
                      "MenuProps": Object {
                        "className": undefined,
                      },
                    }
                  }
                  disabled={true}
                  helperText="Selecione o Ano"
                  id="year-open-select"
                  label="Período"
                  margin="normal"
                  required={false}
                  select={true}
                  value=""
                  variant="standard"
                />
              </WithStyles(Grid)>,
            ],
            "container": true,
            "spacing": 16,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TextField
                  SelectProps={
                    Object {
                      "MenuProps": Object {
                        "className": undefined,
                      },
                    }
                  }
                  helperText="Selecione o Artista"
                  id="artist-open-select"
                  label="Artistas"
                  margin="normal"
                  required={false}
                  select={true}
                  value=""
                  variant="standard"
                />,
                "item": true,
                "lg": 3,
                "md": 3,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "SelectProps": Object {
                    "MenuProps": Object {
                      "className": undefined,
                    },
                  },
                  "children": undefined,
                  "className": undefined,
                  "helperText": "Selecione o Artista",
                  "id": "artist-open-select",
                  "label": "Artistas",
                  "margin": "normal",
                  "onChange": undefined,
                  "required": false,
                  "select": true,
                  "value": "",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TextField
                  SelectProps={
                    Object {
                      "MenuProps": Object {
                        "className": undefined,
                      },
                    }
                  }
                  disabled={true}
                  helperText="Selecione o Ano"
                  id="year-open-select"
                  label="Período"
                  margin="normal"
                  required={false}
                  select={true}
                  value=""
                  variant="standard"
                />,
                "item": true,
                "lg": 3,
                "md": 3,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "SelectProps": Object {
                    "MenuProps": Object {
                      "className": undefined,
                    },
                  },
                  "children": undefined,
                  "className": undefined,
                  "disabled": true,
                  "helperText": "Selecione o Ano",
                  "id": "year-open-select",
                  "label": "Período",
                  "margin": "normal",
                  "onChange": undefined,
                  "required": false,
                  "select": true,
                  "value": "",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
